File: src/ElderCareScheduling.API/Models/Entities/Medication.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ElderCareScheduling.API.Models.Entities;

public class Medication
{
    [Key]
    public Guid Id { get; set; }

    [Required]
    public Guid ElderId { get; set; }

    [ForeignKey(nameof(ElderId))]
    public virtual Elder? Elder { get; set; }

    [Required]
    [MaxLength(200)]
    public string DrugName { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? GenericName { get; set; }

    [MaxLength(100)]
    public string? Specification { get; set; }

    [MaxLength(100)]
    public string? Dosage { get; set; }

    [MaxLength(200)]
    public string? Frequency { get; set; }

    [MaxLength(100)]
    public string? AdministrationRoute { get; set; }

    [MaxLength(500)]
    public string? UsageInstructions { get; set; }

    public DateTime? StartDate { get; set; }

    public DateTime? EndDate { get; set; }

    [MaxLength(500)]
    public string? PrescribingDoctor { get; set; }

    [MaxLength(1000)]
    public string? Precautions { get; set; }

    [MaxLength(500)]
    public string? SideEffects { get; set; }

    public int? RemainingQuantity { get; set; }

    [MaxLength(200)]
    public string? StorageConditions { get; set; }

    public bool IsActive { get; set; } = true;

    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    [Required]
    [MaxLength(50)]
    public string CreatedBy { get; set; } = string.Empty;

    public DateTime? UpdatedAt { get; set; }

    [MaxLength(50)]
    public string? UpdatedBy { get; set; }
}

File: backend/app/schemas/medication.py
from datetime import date, datetime, time
from pydantic import BaseModel
from typing import Optional


class MedicationBase(BaseModel):
    drug_name: str
    generic_name: Optional[str] = None
    dosage: str
    frequency: str
    route: str
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    prescribing_doctor: Optional[str] = None
    pharmacy: Optional[str] = None
    purpose: Optional[str] = None
    side_effects: Optional[str] = None
    notes: Optional[str] = None
    status: str = "active"


class MedicationCreate(MedicationBase):
    elder_id: int


class MedicationUpdate(BaseModel):
    drug_name: Optional[str] = None
    generic_name: Optional[str] = None
    dosage: Optional[str] = None
    frequency: Optional[str] = None
    route: Optional[str] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    prescribing_doctor: Optional[str] = None
    pharmacy: Optional[str] = None
    purpose: Optional[str] = None
    side_effects: Optional[str] = None
    notes: Optional[str] = None
    status: Optional[str] = None


class MedicationResponse(MedicationBase):
    id: int
    elder_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
